### PR TITLE
Android JS Engine selection

### PR DESCRIPTION
**Pull request by CedricGuillemet**
npm installation of JS engine
V8/JSC switch in Gradle
android pipelines for v8 and JSC
build instructions update in an upcoming PR

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -195,24 +195,43 @@ jobs:
...
-      mkdir Dependencies/v8/include
-      mkdir Apps/Playground/Android/app/src/main/jniLibs
-      cp -r node_modules/v8-android/dist/include Dependencies/v8
-      cp -r arm64-v8a Apps/Playground/Android/app/src/main/jniLibs
-      cp -r armeabi-v7a Apps/Playground/Android/app/src/main/jniLibs
-      cp -r x86 Apps/Playground/Android/app/src/main/jniLibs
-      cp -r x86_64 Apps/Playground/Android/app/src/main/jniLibs
-    displayName: 'Install V8 npm'    
+      cd Apps/Playground/Android
+      npm install
```
Comment (syntheticmagus) [inline azure-pipelines.yml]: nit: Is it possible/advisable to get Gradle to do this part, too?

File: Apps/Playground/Android/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (bghgary) [inline Apps/Playground/Android/package-lock.json]: update .gitignore to ignore node_modules?

File: Apps/Playground/Android/app/CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,16 @@ set(BABYLON_NATIVE_PLATFORM "Android")
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
+if (${ANDROID_JSENGINE_LIBNAME} STREQUAL "jsc")
+    set(NAPI_JAVASCRIPT_ENGINE "JavaScriptCore" CACHE STRING "JavaScript engine for N-API.")
+else()
+    set(NAPI_JAVASCRIPT_ENGINE "V8" CACHE STRING "JavaScript engine for N-API.")
+endif()
```
Comment (bghgary) [inline Apps/Playground/Android/app/CMakeLists.txt]: nit: might be better to have an else case that throws an error if the libname is not jsc or v8android.

File: Apps/Playground/Android/package-lock.json (hunk 2 above)
Comment (bghgary) [reply Apps/Playground/Android/package-lock.json]: also remove Dependenceis/v8 from root .gitignore

File: azure-pipelines.yml (hunk 1 above)
Comment (CedricGuillemet) [reply azure-pipelines.yml]: I tried without success to do it with gradle using shell command. No idea why. I'm adding a issue: https://github.com/BabylonJS/BabylonNative/issues/197

**Pull request merged**